src/components/RestaurantItem.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import styled from 'styled-components'
+import { Restaurant } from "../types/Restaurant";
+import Modal from "./Modal";
+import { convertImage } from "../utils/image";
+interface RestaurantItemProps {
+  restaurant: Restaurant;
+}
+
+interface RestaurantItemState {
+  modalOpen: boolean;
+}
+
+class RestaurantItem extends React.Component<RestaurantItemProps, RestaurantItemState> {
+  constructor(props: RestaurantItemProps) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+    };
+  }
+
+  closeModal() {
+    this.setState({ modalOpen: false });
+  }
+
+  render() {
+    const { name, distance, category, description } = this.props.restaurant;
+
+    return (
+      <>
+        <Card onClick={() => this.setState({ modalOpen: true })}>
+          <Favorite>
+          </Favorite>
+          <RestaurantInfo>
+            <CategoryIcon>
+              <img src={convertImage(category)} />
+            </CategoryIcon>
+            <article>
+              <h3>{name}</h3>
+              <p>캠퍼스부터 {distance}분 내</p>
+              <p>{description}</p>
+            </article>
+          </RestaurantInfo>
+        </Card>
+        {
+          this.state.modalOpen && (
+            <Modal restaurant={this.props.restaurant} closeModal={() => this.closeModal()} />
+          )
+        }
```

Comment:
모달의 상태로써 컴포넌트를 그리는 것을 제어하기보다, 모달 내부의 상태를 제어하면 어떨까요?

```suggestion
        <Modal restaurant={this.props.restaurant} isOpen={this.state.modalOpen} closeModal={() => this.closeModal()} />
```